lsp/src/Language/LSP/Server/Core.hs

```diff
@@ -853,18 +856,33 @@ The client can still always choose to cancel for another reason.
 -}
 
 {- Note [Shutdown]
-The 'shutdown' request basically tells the server to clean up and stop doing things.
-In particular, it allows us to ignore or reject all further messages apart from 'exit'.
+The LSP protocol has a two-phase shutdown sequence:
```

Comment:
```suggestion
{- Note [Shutdown]
~~~~~~~~~~~~~~~~~~
The LSP protocol has a two-phase shutdown sequence:
```

While you are here, could you turn this into a note that I can jump to via "Goto Definition"? Like this, the `hls-notes-plugin` can find it.